ruth/data/map.py

```diff
@@ -65,11 +76,46 @@ def name(self):
         """Name of the map."""
         return self.border.name
 
+    def init_current_speeds(self):
+        speeds = nx.get_edge_attributes(self.network, name='speed_kph')
+        travel_times = {}
+        for key, value in speeds.items():
+            travel_times[key] = float(self.segment_lengths[key]) * 3.6 / float(value)
+        nx.set_edge_attributes(self.network, values=speeds, name="current_speed")
+        nx.set_edge_attributes(self.network, values=travel_times, name="current_travel_time")
+
+    def update_current_speeds(self, segment_ids, speeds):
+        new_speeds = {}
+        new_travel_times = {}
+        for (node_from, node_to), speed in zip(segment_ids, speeds):
+            new_speeds[(node_from, node_to)] = speed
+            travel_time = float('inf') if speed == 0 else float(self.segment_lengths[(node_from, node_to)]) * 3.6 / float(speed)
+            new_travel_times[(node_from, node_to)] = travel_time
+        nx.set_edge_attributes(self.simple_network, values=new_speeds, name='current_speed')
+        nx.set_edge_attributes(self.simple_network, values=new_travel_times, name='current_travel_time')
+
+    def get_segment_max_speed(self, node_from, node_to):
+        return self.simple_network[node_from][node_to]['speed_kph']
+
     def set_data_dir(self, path):
         if self.data_dir is not None:
             cl.warn(f"The data dir has changed from '{self.data_dir}' to '{path}.")
         self.data_dir = path
 
+    def osm_route_to_py_segments(self, osm_route: Route) -> List[Segment]:
+        """Prepare list of segments based on route."""
+        result = []
+        for u, v in zip(osm_route[:-1], osm_route[1:]):
+            # if there are parallel edges between two nodes, select the one with the
```

Comment:
There should be no parallel edges, because `self.simple_network` is already a `DiGraph`, not a `MultiDiGraph`, right? Why was this added?

Reply:
Ah yes. My bad. The comment is not necessary here. The functionality was taken from the get_route_edge_attributes since that function is getting deprecated (and the suggested alternative is not useful for our usecase),